### PR TITLE
docs: add plugin limits and lifecycle sections

### PR DESCRIPTION
**Pull request by calvn**
This PR adds a new section to our [Vault Limits and Maximums](https://www.vaultproject.io/docs/internals/limits) page and a new section on our [Plugin System](https://www.vaultproject.io/docs/internals/plugins) internals page. The former is for highlighting resource consumption of external plugins, and the latter is for documenting the lifecycle of a plugin process.

### CODE REVIEW

File: website/content/docs/internals/limits.mdx [hunk 1]
```diff
@@ -285,3 +285,21 @@ The current number of unexpired leases can be monitored via the
...
+initiated by Vault that communicates over RPC. For each secret engine and auth
+method that's enabled as an external plugin, Vault will spawn a process on the
+host system. For the Database Secrets Engines, external database plugins will
+spawn a process for every configured connection.
+
+Regardless of plugin type, each of these processes will incur resource overhead
+on the system, including but not limited to resources such as CPU, memory,
+networking, and file descriptors. There's no specific limit on the number
+secrets engines, auth methods, or database configured connections that can be
+enabled as this depends on the particular plugin resource utilization, the
```
Comment (pcman312) [inline website/content/docs/internals/limits.mdx:301]:
```suggestion
enabled. This ultimately depends on the particular plugin resource utilization, the
```

File: website/content/docs/internals/limits.mdx [hunk 2]
```diff
@@ -285,3 +285,21 @@ The current number of unexpired leases can be monitored via the
...
+extent to which that plugin is being called, and the available resources on the
+system. In general, for plugins of the same type, each additional process
```
Comment (pcman312) [inline website/content/docs/internals/limits.mdx:303]: Wordsmithing suggestion:
```
For plugins of the same type, each additional process will incur a roughly
linear increase in resource utilization. This assumes the usage of each plugin
of the same type is similar.
```

File: website/content/docs/internals/plugins.mdx [hunk 3]
```diff
@@ -37,6 +37,26 @@ It is possible to enable a custom plugin with a name that's identical to a
 built-in plugin. In such a situation, Vault will always choose the custom plugin
 when enabling it.
 
+## Plugin Lifecycle
+
+Vault plugins are long-running processes that remain running once they are
+spawned by Vault, the parent process. Plugin processes can be started by Vault's
+active node and performance standby nodes. Additionally, there are cases where
+plugin processes may be terminated by Vault. These cases include the following:
```
Comment (pcman312) [inline website/content/docs/internals/plugins.mdx:45]:
```suggestion
plugin processes may be terminated by Vault. These cases include but
are not limited to:
```

File: website/content/docs/internals/plugins.mdx [hunk 4]
```diff
@@ -37,6 +37,26 @@ It is possible to enable a custom plugin with a name that's identical to a
...
+
+- Vault active node step-down
+- Vault barrier seal
+- Vault graceful shutdown
+- Database configured connection update
+- Database root credentials rotation
+
+The lifecycle of plugin processes should be managed by automatically and
```
Comment (pcman312) [inline website/content/docs/internals/plugins.mdx:53]: Some more suggested wordsmithing:
```
The lifecycle of plugin processes is managed automatically by Vault. Termination
of these processes is typical in certain scenarios (such as those listed above).
Vault will start plugin processes when needed, typically by lazily loading the
plugin when a request that requires the plugin is received by Vault. A plugin
process may be started or terminated through other internal processes
within Vault as well.
```

Comment (pcman312) [reply website/content/docs/internals/plugins.mdx:53]: Should we also include a note about how a plugin processes should not be terminated by anything other than Vault?

**Review by pcman312 — APPROVED** [review]
Looks good overall. Just a couple of wordsmithing suggestions

File: website/content/docs/internals/limits.mdx (hunk 2 above)
Comment (calvn) [reply website/content/docs/internals/limits.mdx]: Added in d1979de.

File: website/content/docs/internals/plugins.mdx (hunk 3 above)
Comment (calvn) [reply website/content/docs/internals/plugins.mdx]: Updated in d1979de.

File: website/content/docs/internals/plugins.mdx (hunk 4 above)
Comment (calvn) [reply website/content/docs/internals/plugins.mdx:58]: Updated in d1979de, and also added a note on out-of-band termination.

Comment (pcman312) [reply website/content/docs/internals/plugins.mdx:58]: What do you think about changing the first sentence to say "...plugin processes are managed by Vault." I'm namely thinking of using less passive language - "is managed" vs "should be managed".

**Review by pcman312 — APPROVED** [review]

File: website/content/docs/internals/limits.mdx [hunk 5]
```diff
@@ -285,3 +285,21 @@ The current number of unexpired leases can be monitored via the
 The Transform secret engine obeys the [FF3-1 minimum and maximum sizes
 on the length of an input](/docs/secrets/transform#input-limits), which
 are a function of the alphabet size.
+
+### Plugin limits
```
Comment (kalafut) [inline website/content/docs/internals/limits.mdx:289]: If this entire section applies only to external plugins, then perhaps the section should be called "External plugin limits"? Currently the applicability is sort of buried.

Comment (calvn) [reply website/content/docs/internals/limits.mdx]: Updated the section title on cde03f0.

**Review by kalafut — APPROVED** [review]

**Pull request merged**